src/repr/src/row.rs

```diff
@@ -1876,16 +1888,69 @@ impl RowPacker<'_> {
     where
         F: FnOnce(&mut RowPacker) -> R,
     {
-        self.row.data.push(Tag::List.into());
+        // First, assume that the list will fit in 255 bytes, and thus the length will fit in
+        // 1 byte. If not, we'll fix it up later.
         let start = self.row.data.len();
-        // write a dummy len, will fix it up later
-        self.row.data.extend_from_slice(&[0; size_of::<u64>()]);
+        self.row.data.push(Tag::ListTiny.into());
+        // Write a dummy len, will fix it up later.
+        self.row.data.push(0);
 
         let out = f(self);
 
-        let len = u64::cast_from(self.row.data.len() - start - size_of::<u64>());
-        // fix up the len
-        self.row.data[start..start + size_of::<u64>()].copy_from_slice(&len.to_le_bytes());
+        let len = self.row.data.len() - start - 1 - 1;
```

Comment:
Why do we have `- 1 - 1`?

Reply:
1 byte for the tag and 1 byte for the len? Might be helpful to add a quick comment.